text/3723-static-function-argument-unpacking.md

```diff
@@ -0,0 +1,753 @@
+- Feature Name: `static_function_argument_unpacking`
+- Start Date: 2024-10-30
+- RFC PR: [rust-lang/rfcs#3723](https://github.com/rust-lang/rfcs/pull/3723)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC adds call-site unpacking of tuples, tuple structs, and fixed-size arrays, using `...expr` within the function call's parentheses as a shorthand for passing arguments. The full contents of these collections with known sizes are unpacked directly as the next arguments of a function call, desugaring into the corresponding element accesses during compilation.
+
+# Motivation
+[motivation]: #motivation
+
+Argument unpacking reduces the verbosity and increases the ergonomics of Rust, it:
+
+- Improves code writing ergonomics by removing the need for repetitive, unneeded intermediate steps.
+- Allows more concise code in terms of number of lines and, occasionally, line length.
```

Comment:
You need something that constructs the tuple to begin with. Why would we coincidentally have a tuple that happens to have the same order of items as another parameter? If we actually happen to have such a tuple, shouldn't it already be a type (e.g. `struct Vec3 { x: f32, y: f32, z: f32 }`), and shouldn't the called function accept such type?